src/Parser/RawBody.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace React\Http\Parser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\Request;
+
+class RawBody implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    protected $buffer = '';
+    protected $contentLength;
+
+    public function __construct(Request $request)
+    {
+        $headers = $request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        $this->contentLength = $headers['content-length'];
```

Comment:
Take a look here: https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.13

> Any Content-Length greater than or equal to zero is a valid value. Section 4.4 describes how to determine the length of a message-body if a Content-Length is not given.

which refers to 4.4

> If a request contains a message-body and a Content-Length is not given, the server SHOULD respond with 400 (bad request) if it cannot determine the length of the message, or with 411 (length required) if it wishes to insist on receiving a valid Content-Length.

Pretty straightforward.  :) So instead of throwing a exception internal, we should send a response immediately.